### PR TITLE
dynamically generate CPU counts for metrics

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

This changes the behavior of the process and CPU libbeat metrics libraries so that we don't "store" CPU values, in the case of users running this inside VMs were CPU counts can change during runtime. I'm a tad worried about any potential performance impacts of hitting the `NumCPU` call, but unless it's serious, it might be better than coming up with something more sophisticated that refreshes the value every `n` minutes or something.

## Why is it important?

If users are running metricbeat inside a VM where logical CPUs can be changed live, metricbeat can report incorrect metrics.

## Checklist

- [X] My code follows the style guidelines of this project
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- https://github.com/elastic/beats/issues/22885

### CODE REVIEW

File: CHANGELOG-developer.next.asciidoc [hunk 1]
```diff
@@ -52,6 +52,7 @@ The list below covers the major changes between 7.0.0-rc2 and master only.
 - Replace `ACKCount`, `ACKEvents`, and `ACKLastEvent` callbacks with `ACKHandler` and interface in `beat.ClientConfig`. {pull}19632[19632]
 - Remove global ACK handler support via `SetACKHandler` from publisher pipeline. {pull}19632[19632]
 - Make implementing `Close` required for `reader.Reader` interfaces. {pull}20455[20455]
+- Remove `NumCPU` as clients should update the CPU count on the fly in case of config changes in a VM. {pull}23154[23154]
```
Comment (kaiyan-sheng) [inline CHANGELOG-developer.next.asciidoc:55]: Do we consider this a breaking change or bug fix?

**Review by kaiyan-sheng — APPROVED** [review]

File: CHANGELOG-developer.next.asciidoc (hunk 1 above)
Comment (fearful-symmetry) [reply CHANGELOG-developer.next.asciidoc:55]: It's a publicly exported variable, so _technically_ a breaking fix.

**Comment by fearful-symmetry** [comment]
jenkins, test this

File: CHANGELOG-developer.next.asciidoc (hunk 1 above)
Comment (urso) [reply CHANGELOG-developer.next.asciidoc:55]: Although it is a change in reporting, most users will not notice. Users that will notice have had wrong data. Let's call it a fix. How about: `Fix CPU usage metrics on VMs with dynamic CPU assignment. ...`

**Comment by urso** [comment]
Can you please add an entry to Changelog.next.asciidoc as well? The change is clearly a user-visible fix.

**Pull request merged**